runtime/lib/ttnn/operations/context/get_device.cpp

```diff
@@ -8,16 +8,50 @@
 #include "ttmlir/Target/TTNN/program_generated.h"
 
 namespace tt::runtime::ttnn::operations::context {
+static std::unique_ptr<::ttnn::MeshDeviceView>
+constructMeshView(const ::ttnn::MeshDevice &meshDevice,
+                  const ::tt::target::Dim2d *meshViewShape) {
+
+  assert(meshViewShape->y() == 1 &&
+         "Expected 1xN mesh shape for get device op");
+  std::vector<::ttnn::Device *> allDevices = meshDevice.get_devices();
+  std::sort(allDevices.begin(), allDevices.end(),
+            [](const ::ttnn::Device *a, const ::ttnn::Device *b) {
+              return a->id() < b->id();
+            });
+  std::vector<::ttnn::Device *> borrowedDevices(
+      allDevices.begin(), allDevices.begin() + meshViewShape->x());
+  ::ttnn::MeshDeviceView::CoordinateMapper mapper =
+      [&meshViewShape](
+          int deviceId) -> std::optional<::tt::tt_metal::Coordinate> {
+    assert(deviceId < meshViewShape->x() &&
+           "Device ID out of bounds for mesh view");
+    ::tt::tt_metal::Coordinate coord;
+    coord.row = 0;
+    // since we are expecting 1xN mesh shape
+    coord.col = deviceId;
+    return std::make_optional(coord);
+  };
+  return std::make_unique<::ttnn::MeshDeviceView>(borrowedDevices, mapper);
```

Comment:
@jnie-TT, just double checking that we are allowed to use the API in this way.  Perhaps we should have Joseph take a quick look?

Reply:
@nsmithtt Sounds good, I'll sync with Joseph offline to get his input on this.